hack/lib/util.sh

```diff
@@ -504,6 +504,16 @@ kube::util::has_changes_against_upstream_branch() {
   local -r git_branch=$1
   local -r pattern=$2
   local -r not_pattern=${3:-totallyimpossiblepattern}
+  
+  # check prow-aware pattern first
+  if [[ -n ${PULL_BASE_SHA:-} ]]; then
```

Comment:
Might want to move this into its own `kube::util:has_changes_against_pull_base_sha` or something?

Reply:
pulled to its own function and added a wrapper